store/details_v2.go

```diff
@@ -349,6 +354,72 @@ func infoFromStoreSnap(d *storeSnap) (*snap.Info, error) {
 	return info, nil
 }
 
+func componentFromStoreResource(r storeResource) (*snap.Component, error) {
+	typeString := strings.TrimPrefix(r.Type, "component/")
+
+	// nothing was trimmed, so the type must not be a component
+	if typeString == r.Type {
+		return nil, fmt.Errorf("resource is not a component: %s", r.Type)
+	}
+
+	compType, err := snap.ComponentTypeFromString(typeString)
+	if err != nil {
+		return nil, err
+	}
+
+	comp := &snap.Component{
+		Name:        r.Name,
+		Summary:     r.Description.Clean(),
```

Comment:
Should this be empty if not provided by the store? iirc correctly there are some restrictions on the summary length that would not be applied here.

As a more generic thing, are we validating the data we get from the store in some way?